Shopify/cli
# Validation that port used for receiving autentication token from brow…

### WHY are these changes introduced?

Fixes #176 

Along the authorization process the CLI opens an http server to listen to the browser response with the identity token. If for any reason the fixed port used for the server '3456' is already in use, the CLI gets stuck waiting for the response so the process never finishes.

### WHAT is this pull request doing?
- Adds a validation before opening the browser to get the identity token. In case the port is in use the CLI, it prompts the user to terminate the process that is blocking it. In case, the user confirm the action, the process is killed and the authorization goes on opening the browser. On the other hand, if the user rejects finishing the process, the CLI closes gracefully without any error message

<img width="1008" alt="cancel_process_prompt" src="https://user-images.githubusercontent.com/105213827/182577069-c84c117f-c7dd-406f-9712-9b2e6034118a.png">


- Added dependency [kill-port-process](https://github.com/hilleer/kill-port-process)
- Added dependency [find-process](https://github.com/yibn2008/find-process)

### How to test your changes?

1. Logout to ensure the needing of executing the authorization process `yarn shopify auth logout`
2. Run the dev command `yarn shopify app dev`
3. Suspend the process (CTRL + Z) when the CLI is waiting for the browser response
4. Execute again the dev command `yarn shopify app dev`

In MacOS to kill the process that listen to a specific port use follow these steps:
1. `lsof -i -P | grep LISTEN | grep 3456`
2. `kill -9 <pid_from_lsof>`
